### PR TITLE
Linters

### LINKED ISSUE #608: Create Article: Code Linting in JavaScript
<h3 align="center">Catch Before it Crash</h3>

Write an article about **JavaScript code linting** and various popular linting tools available. :white_check_mark:

**Bonus :100: :** Write about other linters such as HTML, CSS, JSON, etc.
<br>
<br>

> ----
<p align="center">Read <a href="https://github.com/FreeCodeCamp/wiki/blob/master/.github/CONTRIBUTING.md"><b>CONTRIBUTING</b></a> Guidelines<br>Chat with us in <kbd><a href="https://gitter.im/FreeCodeCamp/Wiki"> <b>FreeCodeCamp/Wiki</b></a></kbd></p>

> ----

### PR DESCRIPTION
**Pull request by tanham**
Closes #608

### CODE REVIEW

File: JS-Code-Linting-In-Javascript.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Linting is the process of running a program that analyzes your code for programmatic and stylistc errors. A linting tool or a linter marks or flags any potential errors in your code such as syntax errors or incorrectly spelled variable names. This can save time and help you write better code.
```
Comment (atjonathan) [inline JS-Code-Linting-In-Javascript.md]: ~~stylistc~~ stylistic

~~tool or a linter marks~~ tool, or a linter, marks

File: JS-Code-Linting-In-Javascript.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Linting is the process of running a program that analyzes your code for programmatic and stylistc errors. A linting tool or a linter marks or flags any potential errors in your code such as syntax errors or incorrectly spelled variable names. This can save time and help you write better code. 
+
+Linters can be installed using npm or another package manager. They are used from the command line by passing in files, are available as plugins for tools, or are integrated into editors.
```
Comment (atjonathan) [inline JS-Code-Linting-In-Javascript.md]: ~~npm~~ `npm`

**Comment by tanham21** [comment]
hey it's my first time contributing. Should I add the changes in another commit or is there some other way of doing it?

**Comment by atjonathan** [comment]
@tanham21 Yeah, just make a new commit on your PR branch!

**Comment by tanham21** [comment]
I attempted to squash my commits via command line. Sorry if I didn't follow proper protocol. I will next time.

**Comment by atjonathan** [comment]
@tanham21 Push your updated commits using `git push -f origin linters`

**Comment by tanham21** [comment]
Okay. I did that. Did it work?

**Comment by Rafase282** [comment]
It didn't work, try again but make sure to do `d` for drop on the commit at the bottom which should be the one you didnt edit.

**Comment by tanham21** [comment]
I tried again with no luck. My branch 'linters' is not being recognized. Can I start a new pull request or do something else to make this work. Sorry for the inconvenience. 

here's a screenshot of the message I got in command line

screen shot 2016-05-29 at 4 17 03 pm

**Comment by Rafase282** [comment]
`git push -f origin/linters` is what you should type. Feel free to come to the chat and I can help yu faster.

File: JS-Code-Linting-In-Javascript.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Linting is the process of running a program that analyzes your code for programmatic and stylistic errors. A linting tool, or a linter, marks or flags any potential errors in your code such as syntax errors or incorrectly spelled variable names. This can save time and help you write better code. 
+
+Linters can be installed using `npm` or another package manager. Linters can be used from the command line by passing in files. Linters are also available as plugins for tools and sometimes they are directly integrated into editors.
```
Comment (atjonathan) [inline JS-Code-Linting-In-Javascript.md]: Repetitive use of the word `Linters`.

**Comment by atjonathan** [comment]
@Rafase282 Does the filename look ok?

> `JS-Code-Linting-In-Javascript.md`

**Comment by abhisekp** [comment]
@tanham21 
**File rename** 
```diff
- JS-Code-Linting-In-Javascript.md
+ JS-Linters.md
```

File: JS-Code-Linting-In-Javascript.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+Linting is the process of running a program that analyzes your code for programmatic and stylistic errors. A linting tool, or a linter, marks or flags any potential errors in your code such as syntax errors or incorrectly spelled variable names. This can save time and help you write better code.
```
Comment (abhisekp) [inline JS-Code-Linting-In-Javascript.md]: Add a heading 1 at the top

**Comment by abhisekp** [comment]
@tanham21 If possible, please add description about individual linters and their benefits or disadvantages over other linters.

Sort the linters based on popularity and if possible, add other linters such for JSON, HTML, CSS, Sass, TypeScript, CoffeeScript, PHP and other such technologies.

**Comment by atjonathan** [comment]
@abhisekp Shouldn't the file be called:

```diff
- JS-Code-Linting-In-Javascript.md
+ Javascript-Linters.md
```

There are other issues #610 #609 for linting in other technologies, and therefore they will be tackled in different articles.

**Comment by abhisekp** [comment]
@atjonathan ah! right. Also we need an All Linters linking article.

@tanham21 Please ignore my previous file rename suggestion. This is the new name.
```diff
- JS-Code-Linting-In-Javascript.md
+ JS-Linters.md
```

**Comment by Rafase282** [comment]
@tanham21 Hit met up tomorrow or right and and we can work on this PR.

**Comment by Rafase282** [comment]
Awesome work @tanham21

**Pull request merged**